src/charm.py

```diff
@@ -27,17 +32,24 @@
 from ops.model import ActiveStatus, MaintenanceStatus, SecretNotFoundError, WaitingStatus
 
 from resource_manager.gateway import GatewayResourceDefinition, GatewayResourceManager
-from resource_manager.resource_manager import InvalidResourceError
+from resource_manager.http_route import (
+    HTTPRouteResourceDefinition,
+    HTTPRouteResourceManager,
+    HTTPRouteType,
+)
 from resource_manager.secret import SecretResourceDefinition, TLSSecretResourceManager
+from resource_manager.service import ServiceResourceDefinition, ServiceResourceManager
 from state.config import CharmConfig
 from state.gateway import GatewayResourceInformation
+from state.http_route import HTTPRouteResourceInformation
 from state.tls import TLSInformation
 from state.validation import validate_config_and_integration
 from tls_relation import TLSRelationService, get_hostname_from_cert
 
 TLS_CERT = "certificates"
 logger = logging.getLogger(__name__)
 CREATED_BY_LABEL = "gateway-api-integrator.charm.juju.is/managed-by"
+INGRESS_RELATION = "gateway"
```

Comment:
For coherence shouldn't `TLS_CERT` be named `TLS_CERT_RELATION` and both declared next to each other ?

Reply:
I renamed it and reordered the constants